### PR TITLE
Fix unresolved links in documentation

### LINKED ISSUE #957: Unresolved links in documentation
We seem to have unresolved links in the documentation, that are generating warnings in our CI:
https://github.com/boa-dev/boa/actions/runs/408553547

### PR DESCRIPTION
**Pull request by tofpie**
This Pull Request fixes #957.

It changes the following:
- Unresolved or misinterpreted links in the documentation

### CODE REVIEW

File: boa/src/syntax/ast/op.rs [hunk 1]
```diff
@@ -680,8 +680,8 @@ pub enum LogOp {
     ///  - [ECMAScript reference](
     ///  - [MDN documentation][mdn]
     ///
-    /// [spec]: https://tc39.es/ecma262/#prod-LogicalORExpression)
-    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_OR
+    /// [spec]: <https://tc39.es/ecma262/#prod-LogicalORExpression)>
+    /// [mdn]: <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_OR>
```
Comment (Razican) [inline boa/src/syntax/ast/op.rs:684]: I believe this is not correct, right? It should show `[spec]` next to `[ECMAScript reference]`, and then, the links should not have `<...>`. Also, the `spec` link should not have a closing `)`

File: boa/src/value/mod.rs [hunk 2]
```diff
@@ -247,8 +247,8 @@ impl Value {
     /// For scalar types it should be false, for objects check the private field for extensibilaty.
     /// By default true.
     ///
-    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal would turn extensible to false/>
-    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze would also turn extensible to false/>
+    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal> would turn extensible to false
+    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze> would also turn extensible to false
```
Comment (Razican) [inline boa/src/value/mod.rs:251]: I believe it's best to treat variables and variables as code, now that we are changing this.

```suggestion
    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/seal> would turn `extensible` to `false`
    /// <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze> would also turn `extensible` to `false`
```

**Review by Razican — COMMENTED** [review]
Thanks for the contribution! This cleans up a lot of stuff in the docs :)

I only noticed two minor things that could be improved.

**Review by Razican — APPROVED** [review]
Thanks! This looks good to me :)

**Review by RageKnify — APPROVED** [review]

**Pull request merged**